optuna/samplers/_cmaes.py

```diff
@@ -216,6 +233,25 @@ def __init__(
                 ExperimentalWarning,
             )
 
+        if self._source_trials is not None:
+            warnings.warn(
+                "`source_trials` option is an experimental feature."
+                " The interface can change in the future.",
+                ExperimentalWarning,
+            )
+
+        if source_trials is not None and (x0 is not None or sigma0 is not None):
+            raise ValueError(
+                "It is prohibited to pass `source_trials` argument when "
+                "x0 or sigma0 is specified."
+            )
+
+        # TODO(c-bata): Support WS-sep-CMA-ES
+        if source_trials is not None and use_separable_cma:
+            raise ValueError(
+                "It is prohibited to pass `source_trials` argument when " "using separable CMA-ES."
+            )
```

Comment:
How about adding test cases to confirm that the argument checking works as expected?